lib/AST/NameLookup.cpp

```diff
@@ -1067,6 +1068,13 @@ void MemberLookupTable::addMember(Decl *member) {
   // Add this declaration to the lookup set under its compound name and simple
   // name.
   vd->getFullName().addToLookupTable(Lookup, vd);
+
+  // Add this declaration to the lookup set under any @_implements name as well
```

Comment:
I think the right place to hook this up is not inside name lookup, but in lib/Sema/TypeCheckProtocol.cpp, in particular `lookupValueWitnesses()`. When you begin conformance checking, maybe walk the members to collect all _implements attributes, building a mapping from requirement to witness inside the conformance. Then `lookupValueWitness()` can avoid most of the work by checking the pre-populated map first.